diff --git a/bin/qtile b/bin/qtile
index e7c79696..2ddbe7cd 100755
--- a/bin/qtile
+++ b/bin/qtile
@@ -33,4 +33,5 @@ import libqtile.core.lifecycle  # noqa: F401, E402
 from libqtile.scripts.main import main  # noqa: E402
 
 if __name__ == '__main__':
+    __import__("importlib").import_module("libqtile.utils").restore_os_environment()
     main()
diff --git a/libqtile/utils.py b/libqtile/utils.py
index 08ac2523..8b2ad126 100644
--- a/libqtile/utils.py
+++ b/libqtile/utils.py
@@ -280,6 +280,12 @@ def guess_terminal(preference=None):
 
     logger.error('Default terminal has not been found.')
 
+def restore_os_environment():
+    pythonpath = os.environ.pop("QTILE_SAVED_PYTHONPATH", "")
+    os.environ["PYTHONPATH"] = pythonpath
+    path = os.environ.pop("QTILE_SAVED_PATH", None)
+    if path:
+        os.environ["PATH"] = path
 
 def scan_files(dirpath, *names):
     """
